### DISCUSSION TITLE
Callouts should not use ## in first line

### DISCUSSION THREAD

**Discussion by stefansimik** [quarto-dev/quarto-cli#4172] [Q&A] [answered]
Hi 

I would suggest, that callouts should not use ## in first line, because it is showed like header in JupyterLab Table of contents, but the point is not to be a header - it  is is standard parameter like any other.

See this code:

```
::: {.callout-note collapse="true" icon=false}
## Callout header

Content of the callout
:::
```

When we use this in any Markdown cell in Jupyterlab, this is displayed in TOC like Header of Level 2,
despite it is no header. Here we simply ## in a bad way, that represents Header Lvl2, despite it is not in reality.

See here:

![screenshot](https://user-images.githubusercontent.com/6208056/215989370-b2be091e-2e08-48c6-ab35-425c1b16eab5.png)

This does not look so bad, in one small example, but when you use a lot of callouts, this make from TOC in Jupyterlab complete mess
of invalid headers, that do NOT really exist. And this is not problem of the TOC in Jupyterlab, which shows them correctly, but it is problem of misusing ## syntax for bad purpose - and should be replaced by some other characters.

## Solution

We should change those ## used in callout, to some other syntax, that is not conflicting with standard Markdown syntax.
I know, that it is already used, so let's make an alternative syntax, that will be marked as preferred and recommended, despite old syntax will be still working for backward compatibility.

I am also adding the *.ipynb file in screenshot, for easy reproducibility.
(just remove the last *.txt suffix, otherwise I could not upload *.ipynb file)

[Example.ipynb.txt](https://github.com/quarto-dev/quarto-cli/files/10554866/Example.ipynb.txt)

**Comment by dragonstyle** [comment, marked as answer]
I definitely sympathize - first a workaround and then a suggestion for an alternative that I think may be a better long term plan.

1) As a workaround, callouts accept any heading level (not just level 2 headings), so one workaround is merely to use a heading like H6 or something that doesn't appear in the TOC. (Tabs work the same way)

```
::: {.callout-note collapse="true" icon=false}
###### Callout header

Content of the callout
:::
```

2) Alternatively, I think no matter what syntax we select for callouts (and other Quarto specific elements like tabsets or the like), to create a great authoring experience in Jupyter Lab / Notebooks we'll want to extend the environment to provide more intelligent rendering for the elements. This doesn't preclude not using headings, but I think that my main point is that we should not let the authoring tool drive the syntax for elements unless absolutely necessary.

**Reply by stefansimik** [reply]
I completely agree.

First, thank you very much for nice and elegant temporary solution, that works well  👍

**Comment by rutv1k46** [comment]
Is there a way to replicate what @stefansimik posted in their question. I have few callouts which I'm using to show exercises on a html webpage and I would like those callouts to appear in the toc. I'm following the similar syntax for defining the callouts titles:

```
::: {.callout-caution appearance="simple" icon=false}
### Exercise 1

The exercise.
:::
```

**Reply by cderv** [reply]
> I would like those callouts to appear in the toc.

The TOC of the output document, or are you referring to same issue in `ipynb` in Jupyter Lab editor ?

**Reply by rutv1k46** [reply]
The TOC of the output document. I'm working with html format and I would like to have the callouts to appear in the TOC.

**Reply by mcanouil** [reply]
That's not going to happen.
Callouts are similar to paragraph, as such they don't have any reason to be in a table of content.

Note that this thread was about the issue that some markdown parser (from the IDE) detected the callout title and used it in the outline which is an unfortunate side effect of using `##`. This has been answerered already.

**Reply by cderv** [reply]
This is not possible and I am not sure it will be. Headings are not meant to be part of supporting syntax in callout

See this related discussion comments:
- https://github.com/quarto-dev/quarto-cli/issues/6826
